src/OmniSharp.Cake/Configuration/Parser/ConfigurationParser.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+
+namespace OmniSharp.Cake.Configuration.Parser
+{
+    internal sealed class ConfigurationParser
+    {
+        [SuppressMessage("Microsoft.Usage", "CA2202:Do not dispose objects multiple times")]
+        public IDictionary<string, string> Read(string path)
+        {
+            // Make sure that the configuration file exist.
+            if (!File.Exists(path))
+            {
+                throw new FileNotFoundException("Unable to find the configuration file.", path);
+            }
+
+            using (var stream = File.OpenRead(path))
+            using (var reader = new StreamReader(stream))
+            {
+                return ReadFileContent(reader.ReadToEnd());
+            }
+        }
+
+        private static IDictionary<string, string> ReadFileContent(string text)
+        {
+            var tokens = ConfigurationTokenizer.Tokenize(text);
+            var section = string.Empty;
+            var result = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            while (tokens.Current != null)
+            {
+                switch (tokens.Current.Kind)
+                {
+                    case ConfigurationTokenKind.Section:
+                        section = ParseSection(tokens);
+                        break;
+                    case ConfigurationTokenKind.Value:
+                        var pair = ParseKeyAndValue(tokens, section);
+                        result[pair.Key] = pair.Value;
+                        break;
+                    default:
+                        throw new InvalidOperationException("Encountered unexpected token.");
+                }
+            }
+            return result;
+        }
+
+        private static string ParseSection(ConfigurationTokenStream tokens)
+        {
+            var value = tokens.Current.Value;
+            if (ContainsWhiteSpace(value))
+            {
+                throw new InvalidOperationException("Sections cannot contain whitespace.");
```

Comment:
What does cake do in this scenario, does it throw if the config has invalid values or does it default those values?

Reply:
Configuration handling is "copy-pasted-with-pride" from the Cake repo. Now that I look at it, I should remove any exception throwing and just set sensible defaults.

Reply:
A second set of 👀 always helps!

Reply:
I went the Catch-and-Log route instead. Makes more sense than silently continue and keeps code in sync with code in Cake.